### PR TITLE
chore: upgrade coraza to go v1.19 and add v1.21 tests

### PR DESCRIPTION
**Pull request by jptosso**
Go was updated to 1.21. We are keeping compatibility for v1.19, v1.20, v1.21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +47,8 @@ The Coraza Project maintains implementations and plugins for the following serve
 
 ## Prerequisites
 
+* Go v1.19+ or tinygo compiler
+* Linux distribution (Debian or Centos recommended) or Mac. Windows not supported yet.
 * Golang compiler v1.18+
```
Comment (M4tteoP) [inline README.md:52]: Some duplicated lines in the prerequisites, I think it should be
```
* Go v1.19+ or tinygo compiler
* Linux distribution (Debian or Centos recommended), Windows or Mac.
```

**Comment by jcchavezs** [comment]
Are we ready to move forward with this?

**Comment by jcchavezs** [comment]
@jptosso could you update this PR?

**Comment by jptosso** [comment]
> @jptosso could you update this PR?

done

**Review by jcchavezs — APPROVED** [review]

**Pull request merged**